### PR TITLE
Added DHCPv6 support to OSv

### PR DESCRIPTION
**Pull request by cmyers-spirent**
- Added DHCPv6 support to OSv
- Added additional TCP/UDP stats for debug.
  I guess I added TCP/UDP stats to my local repo a while ago for debug and never pushed it.

Was going to clean this up a bit more, but getting some pressure to finish it up....

### CODE REVIEW

File: bsd/porting/networking.cc [hunk 1]
```diff
@@ -369,5 +381,26 @@ std::string if_ip(std::string if_name) {
...
+bool dhcp6_get_if_enable(const std::string &if_name, bool &enable)
+{
+    return false; 
 }
+
+void dhcp6_set_if_stateless(const std::string &if_name, bool enable)
+{
+}
+
+bool dhcp6_get_if_enable(const std::string &if_name, bool &enable)
```
Comment (jmkho-spirent) [inline bsd/porting/networking.cc]: dhcp_get_if_stateless?

File: bsd/porting/networking.hh [hunk 2]
```diff
@@ -26,9 +26,24 @@ namespace osv {
...
+    void dhcp_set_if_enable(const std::string &if_name, bool enable);
+    bool dhcp_get_if_enable(const std::string &if_name, bool &enable);
+
+    // Modules like cloud-init don't have easy way to find if INET6 is enabled
+    // so these functions are always present and stubbed as noop in
+    // networking.cc
+    void dhcp6_set_if_enable(const std::string &if_name, bool enable);
+    bool dhcp6_get_if_enable(const std::string &if_name, bool &enable);
+    void dhcp6_set_if_stateless(const std::string &if_name, bool enable);
+    bool dhcp6_get_if_enable(const std::string &if_name, bool &enable);
```
Comment (jmkho-spirent) [inline bsd/porting/networking.hh]: dhcp6_get_if_stateless?

File: modules/libtools/netstat.cc [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    mib[1] = PF_INET;
+    mib[2] = proto;
+    mib[3] = proto_ctl;
+
+    if (osv_sysctl(mib, 4, NULL, &needed, NULL, 0) < 0) {
+        fprintf(stderr, "PCBLIST sysctl failed to get buffer size\n");
+        return errno;
+    }
+
+    sp = new char[needed];
```
Comment (DerangedMonkeyNinja) [inline modules/libtools/netstat.cc]: Surely you mean `std::vector<char>(needed)`?
At least that way, you don't have to worry about deleting `sp` on exit.

File: bsd/sys/netinet/udp_stat.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
+ */
+
+#ifndef _NETINET_UDP_STAT_H_
+#define	_NETINET_UDP_STAT_H_
```
Comment (DerangedMonkeyNinja) [inline bsd/sys/netinet/udp_stat.h]: nit: weird whitespace?

File: bsd/sys/sys/mbuf.h [hunk 5]
```diff
@@ -877,6 +877,7 @@ u_int		 m_length(struct mbuf *, struct mbuf **);
 int		 m_mbuftouio(struct uio *, struct mbuf *, int);
 void		 m_move_pkthdr(struct mbuf *, struct mbuf *);
 struct mbuf	*m_prepend(struct mbuf *, int, int);
+void             m_print(const struct mbuf *, int);
```
Comment (DerangedMonkeyNinja) [inline bsd/sys/sys/mbuf.h]: nit: whitespace

File: core/dhcp.cc [hunk 6]
```diff
@@ -757,15 +770,34 @@ namespace dhcp {
         // FIXME: free packets and states
     }
 
+    void dhcp_worker::set_if_enable(const std::string &if_name, bool enable)
+    {
+        _if_config[if_name] = enable;
+    }
+
+    bool dhcp_worker::get_if_enable(const std::string &if_name, bool &enable)
```
[the remark this answers is not in the document]
Comment (cmyers-spirent) [reply core/dhcp.cc]: current behavior sets flags in cloud init and through "cmdline" startup options.
If an option has not been set, I would like to apply defaults based off of other values which have been set.  internal storage is map so unset state is when value is not in map.
so nullable bool type would be another way to get behavior.
e.g. if you don't enable any ipv4 and ipv6 options you get both.
       if you enable only ipv4 or ipv6 then you only get option you enabled.
Is there a better way to do this?

**Pull request merged**